tests/components/airpatrol/test_climate.py

```diff
@@ -0,0 +1,416 @@
+"""Test the AirPatrol climate platform."""
+
+from datetime import timedelta
+
+from freezegun.api import FrozenDateTimeFactory
+
+from homeassistant.components.airpatrol.climate import (
+    HA_TO_AP_FAN_MODES,
+    HA_TO_AP_HVAC_MODES,
+    HA_TO_AP_SWING_MODES,
+)
+from homeassistant.components.climate import (
+    ATTR_CURRENT_HUMIDITY,
+    ATTR_CURRENT_TEMPERATURE,
+    ATTR_FAN_MODE,
+    ATTR_FAN_MODES,
+    ATTR_HVAC_MODE,
+    ATTR_HVAC_MODES,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_TEMP,
+    ATTR_SWING_MODE,
+    ATTR_SWING_MODES,
+    DOMAIN as CLIMATE_DOMAIN,
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    SERVICE_SET_FAN_MODE,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_SWING_MODE,
+    SERVICE_SET_TEMPERATURE,
+    SWING_OFF,
+    SWING_ON,
+    HVACMode,
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    CONF_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+)
+from homeassistant.core import HomeAssistant
+
+from tests.common import SnapshotAssertion, async_fire_time_changed
+
+
+async def test_async_setup_entry_adds_entities(
+    hass: HomeAssistant,
+    load_integration,
+    mock_airpatrol_client,
+    get_data,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test async_setup_entry creates and adds AirPatrolClimate entities that have climate data."""
+    unit_with_climate = get_data(unit_id="unit1", name="Unit 1")
+    unit_without_climate = get_data(unit_id="unit2", name="Unit 2", climate=None)
+    mock_airpatrol_client.get_data.return_value = [
+        unit_with_climate[0],
+        unit_without_climate[0],
+    ]
+    await load_integration()
+
+    assert hass.states.get("climate.unit_1") == snapshot
+    assert hass.states.get("climate.unit_2") == snapshot
```

Comment:
instead use the `snapshot_platform` helper

Reply:
The comments would be very helpful if they also included a link to some documentation around the suggested methods. As I've stated this is my first contribution to Home Assistant along with a lack of python experience. I can't find anything on developers.home-assistant.io that indicates how this method would work.